src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RegularExpressionAttribute.cs

```diff
@@ -21,9 +21,16 @@ public RegularExpressionAttribute(string pattern)
             : base(() => SR.RegexAttribute_ValidationError)
         {
             Pattern = pattern;
+            MatchTimeoutInMilliseconds = 2000;
         }
 
         /// <summary>
+        ///     Gets or sets the timeout to use when matching the regular expression pattern (in milliseconds)
+        ///     (-1 means never timeout).
+        /// </summary>
+        public int MatchTimeoutInMilliseconds { get; set; }
```

Comment:
Yes. From the discussions that took place I understood it was OK as we're adding new API not changing old (cc @weshaggard) and this is necessary to address the security issue. But if I need to change any other files to update the contracts please point me to them.